### PR TITLE
Casting fixes

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: src/addressUtils.c [hunk 1]
```diff
@@ -140,14 +140,19 @@ size_t unboxChecksummedAddress(
...
 	const uint8_t* unboxedBuffer;
 	{
 		parseTokenWithValue(&view, CBOR_TYPE_ARRAY, 2);
 		{
 			parseTokenWithValue(&view, CBOR_TYPE_TAG, CBOR_TAG_EMBEDDED_CBOR_BYTE_STRING);
 
-			unboxedSize = parseToken(&view, CBOR_TYPE_BYTES);
+			uint64_t tmp = parseToken(&view, CBOR_TYPE_BYTES);
+			// Validate that we can down-cast
+			STATIC_ASSERT(sizeof(tmp) >= sizeof(SIZE_MAX), "wut?");
```
Comment (refi93) [inline src/addressUtils.c]: this assertion error message could be more comprehensive imho

File: src/assert.h [hunk 2]
```diff
@@ -19,13 +19,13 @@ extern void assert(int cond, const char* msgStr);
...
-    )
+	(sizeof(strLiteral) > size \
+	 ? (strLiteral) + sizeof(strLiteral) - size \
+	 : strLiteral \
+	)
 
 #define _FILE_LINE_ __FILE__ ":" _TO_STR2_(__LINE__)
 
-#define ASSERT(cond) assert(cond, _SHORTEN_( _FILE_LINE_, _MAX_ASSERT_LENGTH_))
+#define ASSERT(cond) assert((cond), _SHORTEN_( _FILE_LINE_, _MAX_ASSERT_LENGTH_))
```
Comment (refi93) [inline src/assert.h:29]: the parentheses around `cond` inside the `assert` are probably redundant

**Comment by refi93** [comment]
I made an additional commit where I made the downcastings that the compiler complained about explicit which silences the warnings and also lets explicitly know future contributors what's going on.

File: src/assert.h (hunk 2 above)
Comment (ppershing) [reply src/assert.h:29]: Hmm, maybe. Explicit parenthesis won't break anything though

File: src/endian.h [hunk 3]
```diff
@@ -34,17 +34,18 @@ inline uint8_t u1be_read(const uint8_t* inBuffer)
 
 inline uint16_t u2be_read(const uint8_t* inBuffer)
 {
-	return ((uint16_t) u1be_read(inBuffer) << 8) | (u1be_read(inBuffer + 1));
+	// bitwise OR promotes types smaller than int to int
+	return (uint16_t) ((u1be_read(inBuffer) << 8) | (u1be_read(inBuffer + 1)));
```
Comment (ppershing) [inline src/endian.h]: Hmm, maybe do
`STATIC_ASSERT(sizeof(uint32_tint) == sizeof(unsigned)` and explicitly cast both sides to unsigned?

File: src/addressUtils.c [hunk 4]
```diff
@@ -140,14 +140,19 @@ size_t unboxChecksummedAddress(
...
+			VALIDATE(tmp < SIZE_MAX, ERR_INVALID_DATA);
```
Comment (ppershing) [inline src/addressUtils.c]: hmm, looking at this, I am not sure about signedness of the result. Can we make explicit `tmp < (uint64_t) SIZE_MAX` casting to make sure left side is not cast to signed? It should not but better be explicit about it

File: src/addressUtils.c (hunk 1 above)
Comment (ppershing) [inline src/addressUtils.c]: maybe just reword to 'bad int size' ?

Comment (refi93) [reply src/addressUtils.c]: yes, that would make sense

File: src/endian.h (hunk 3 above)
Comment (refi93) [reply src/endian.h]: done, although my question is why not to cast directly to uint32_t?

File: src/addressUtils.c (hunk 4 above)
Comment (refi93) [reply src/addressUtils.c]: done

File: src/addressUtils.c (hunk 1 above)
Comment (refi93) [reply src/addressUtils.c]: done

**Review by ppershing — COMMENTED** [review]
I guess we can merge this :-)

**Pull request merged**